l1-contracts/test/foundry/unit/concrete/state-transition/StateTransitionManager/CreateNewChain.t.sol

```diff
@@ -36,4 +36,53 @@ contract createNewChainTest is StateTransitionManagerTest {
         assertEq(newChainAdmin, admin);
         assertNotEq(newChainAddress, address(0));
     }
+
+    function test_SuccessfulCreationOfNewChainAndReturnChainIds() public {
+        createNewChain(getDiamondCutData(diamondInit));
+        createNewChainWithId(getDiamondCutData(diamondInit), 10);
+
+        uint256[] memory chainIds = chainContractAddress.getAllHyperchainChainIDs();
+        assertEq(chainIds.length, 2);
+        assertEq(chainIds[0], chainId);
+        assertEq(chainIds[1], 10);
+    }
+
+    function test_SuccessfulCreationOfNewChainAndReturnChainAddresses() public {
+        createNewChain(getDiamondCutData(diamondInit));
+        createNewChainWithId(getDiamondCutData(diamondInit), 10);
+
+        address[] memory hyperchainAddresses = chainContractAddress.getAllHyperchains();
+        assertEq(hyperchainAddresses.length, 2);
+        assertEq(hyperchainAddresses[0], chainContractAddress.getHyperchain(chainId));
+        assertEq(hyperchainAddresses[1], chainContractAddress.getHyperchain(10));
+    }
+
+    function test_RevertWhen_AlreadyDeployedHyperchainAddressIsZero() public {
+        vm.expectRevert(bytes("STM: hyperchain zero"));
+
+        chainContractAddress.registerAlreadyDeployedHyperchain(chainId, address(0));
+    }
+
+    function test_SuccessfulRegisterAlreadyDeployedHyperchain() public {
+        address randomHyperchain = makeAddr("randomHyperchain");
+
+        chainContractAddress.registerAlreadyDeployedHyperchain(10, randomHyperchain);
+
+        assertEq(chainContractAddress.getHyperchain(10), randomHyperchain);
+    }
+
+    function test_RevertWhen_HyperchainLimitReached() public {
+        for (uint256 i = 0; i < MAX_NUMBER_OF_HYPERCHAINS; i++) {
+            createNewChainWithId(getDiamondCutData(diamondInit), 10 + i);
+        }
+
+        uint256[] memory chainIds = chainContractAddress.getAllHyperchainChainIDs();
+        assertEq(chainIds.length, MAX_NUMBER_OF_HYPERCHAINS);
+
+        vm.stopPrank();
+        vm.startPrank(governor);
```

Comment:
Why do you use this pattern?

Reply:
I used this pattern because I did not want to interfere with the tests that were previously created by someone else. However, I made a small modification and now it should be fine.